src/Marker.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react'
+
+import { MapContext } from './Map'
+
+import {
+  MarkerOptions,
+  createCoordinate,
+  propsToMarkerConstructionOptions,
+} from './utils'
+
+type MarkerProps = {
+  latitude: number
+  longitude: number
+} & MarkerOptions
+
+export const Marker: React.FC<MarkerProps> = ({
+  latitude,
+  longitude,
+  ...options
+}) => {
+  const { mapkit, map } = React.useContext(MapContext)
+  const marker = React.useRef<mapkit.MarkerAnnotation>()
+
+  React.useEffect(() => {
+    if (mapkit && map) {
```

Comment:
It feels like you shouldn't need to check for these things after the component is mounted?

Reply:
Gotta make typescript happy :)